Composer/packages/ui-plugins/select-skill-dialog/src/BeginSkillDialogField.tsx

```diff
@@ -46,6 +96,12 @@ export const BeginSkillDialogField: React.FC<FieldProps> = (props) => {
     },
   };
 
+  const skillIdUiSchema = uiOptions.properties?.id || {};
+  skillIdUiSchema.serializer = {
+    get: (value) => skillsInSettings.get(value),
+    set: (value) => value,
+  };
+
   return (
     <React.Fragment>
       <SchemaField
```

Comment:
I think we should replace this `SchemaField` with a combobox that doesn't set the `id` in the dialog file.